core/src/main/java/org/infinispan/interceptors/impl/TransactionalStoreInterceptor.java

```
@@ -0,0 +1,151 @@
+         return visitSingleStore(ctx, command.getKey());
+      }
+
+      @Override
+      public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
+         CacheEntry entry = ctx.lookupEntry(command.getKey());
+         InternalCacheEntry ice;
+         if (entry instanceof InternalCacheEntry) {
+            ice = (InternalCacheEntry) entry;
+         } else if (entry instanceof DeltaAwareCacheEntry) {
+            AtomicHashMap<?, ?> uncommittedChanges = ((DeltaAwareCacheEntry) entry).getUncommittedChages();
+            ice = entryFactory.create(entry.getKey(), uncommittedChanges, entry.getMetadata(), entry.getLifespan(), entry.getMaxIdle());
+         } else {
+            ice = entryFactory.create(entry);
+         }
+         MarshalledEntryImpl marshalledEntry = new MarshalledEntryImpl(ice.getKey(), ice.getValue(), PersistenceUtil.internalMetadata(ice), marshaller);
+         modifications.addMarshalledEntry(ice.getKey(), marshalledEntry);
+         return null;
+      }
+
+      @Override
+      public Object visitReplaceCommand(InvocationContext ctx, ReplaceCommand command) throws Throwable {
+         return visitSingleStore(ctx, command.getKey());
+      }
+
+      @Override
+      public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) throws Throwable {
+         Map<Object, Object> map = command.getMap();
+         for (Object key : map.keySet())
+            visitSingleStore(ctx, key);
+         return null;
+      }
+
+      @Override
+      public Object visitRemoveCommand(InvocationContext ctx, RemoveCommand command) throws Throwable {
+         modifications.removeEntry(command.getKey());
+         return null;
+      }
+
+      @Override
+      public Object visitClearCommand(InvocationContext ctx, ClearCommand command) throws Throwable {
+         persistenceManager.clearAllStores(SHARED);
```

Since clear is always non-transactional, I think you can ignore `ClearCommand` (or even throw an `UnsupportedOperationException`).